doc/md/guides/testing/testing-migrations.mdx

```diff
@@ -0,0 +1,276 @@
+---
+id: testing-migrations
+title: Testing Migrations
+slug: /guides/testing/migrations
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+Testing your database schema and migrations is crucial to ensure code behaves as expected, catch bugs early,
```

Comment:
Testing migrations is mostly around data migrations. Atlas covers other migrations using linting (is the change safe and compliant) and potentially diffing (does applying this migration leave the db in the state that we intended).  Data migrations are essentially refactorings to the data in the database. If these are done incorrectly and leave the data in the wrong state - this can be very problematic and difficult to recover from (its a non reversible op). For this reason its important to test the logic. Testing this is cumbersome. You need to spin up an empty db, run the migrations up to the migration *before* the one you're testing, seed some test data, run the migration your testing, run assertions to make sure your logic worked.

Atlas migrate test lets you define these cases in a simple concise syntax, and acts as a harness that can run these tests for you during local development and in CI.